### DISCUSSION TITLE
bpaf_cauwugo: How to enable completion for aliases

### DISCUSSION THREAD

**Discussion by Stock84-dev** [pacak/bpaf#157] [Q&A] [answered]
This works fine:
```bash
cauwugo check -p<TAB>
```
But with alias it doesn't:
```bash
alias c="cauwugo check"
c -p<TAB>
```
I use zsh.

**Comment by pacak** [comment, marked as answer]
By default zsh tries to expand the alias before doing the completion, unless told otherwise with `completealiases`, so removing `setopt completealiases` from `.zshrc` or adding `unsetopt completealiases` there should do the trick.

A few related links:

https://zsh.sourceforge.io/Doc/Release/Options.html#index-COMPLETE_005fALIASES
https://unix.stackexchange.com/questions/250314/whats-the-intended-use-case-for-complete-aliases-in-zsh
https://unix.stackexchange.com/questions/335145/using-zsh-autocompletion-for-alias
